tests/resources/test_material_run.py

```diff
@@ -211,3 +211,40 @@ def test_delete_material_run(collection, session):
     assert expected_call == session.last_call
 
 
+def test_build_discarded_objects_in_material_run():
+    # Note:  This is really here just for test coverage - couldn't figure out how to
+    #        get past the validation/serialization in MaterialRun.build - it might just be dead code
```

Comment:
Yeah, the method `_build_list_of_soft_links` is not strictly opinionated about what form the input information comes in, but `build()` does force it to be a `DictSerializable`. 
You could also mark the dead lines as `noqa` if you don't want to include this artificial test.